R/contrs.R

```diff
@@ -0,0 +1,84 @@
+#' Deviation Contrast Matrix
+#'
+#' Build a deviation contrast matrix, a type of _effects contrast_ matrix.
+#'
+#' @inheritParams stats::contr.sum
+#'
+#' @details
+#' In effects coding, unlike dummy coding ([stats::contr.treatment()]), each
+#' contrasts sums to 0. In regressions models, this results in an intercept that
+#' represents the (unweighted) average of the group means. In ANOVA settings,
+#' this also guarantees that lower order effects represent _main_ effects (and
+#' not _simple_ or _conditional_ effects, as is the case when using
+#' [stats::contr.treatment()], which is `R`'s default).
+#' \cr\cr
+#' `contr.deviation`, unlike [stats::contr.sum()], also has the added benefit
+#' that the factor-related coefficients are interpretable. In fact, they
+#' represent the same contrasts as those of [stats::contr.treatment()]: the
+#' difference of each level from the base level.
+#'
+#' @seealso [stats::contr.sum()]
+#'
+#' @examples
+#' \dontrun{
```

Comment:
I get that you want to add comments to the outputs but if for some reason the output changes later, then these "fixed" examples won't be updated, maybe you could add the explanations in comments above/below the output instead?

Reply:
I see no reason the outputs will ever change, and it is really useful to have the comments *on* the relevant output row... :/

Reply:
I don't know, maybe there could be a tiny bug in `stats::contr.treatment()` that is solved in a future R version. It's just to be future-proof. I'm ok with letting the examples like this but maybe you could just add a snapshot test of the output with a comment saying to update these examples if the snapshot changes.